src/query/plan/operator.cpp

```
@@ -2003,7 +2015,16 @@ bool Delete::DeleteCursor::Pull(Frame &frame, ExecutionContext &context) {
   for (TypedValue &expression_result : expression_results) {
     if (MustAbort(context)) throw HintedAbortError();
     if (expression_result.type() == TypedValue::Type::Edge) {
-      auto maybe_value = dba.RemoveEdge(&expression_result.ValueEdge());
+      auto &ea = expression_result.ValueEdge();
+      if (context.auth_checker &&
+          !(context.auth_checker->Accept(*context.db_accessor, ea, auth::FineGrainedPermission::CREATE_DELETE) &&
+            context.auth_checker->Accept(*context.db_accessor, ea.To(), storage::View::NEW,
+                                         auth::FineGrainedPermission::CREATE_DELETE) &&
```

Check with Boris if we need to rather throw an exception here.